### PR TITLE
feat: add eslint-plugin-unicorn

### PR DESCRIPTION
**Pull request by mhatvan**
#### Description of Change
- add eslint-plugin-unicorn
- change from .eslintrc to .eslintrc.js to be able to write comments and use dirname
- restructure and update lint config
- add separate lint:fix script
- autofix all files with unicorn ruleset plus minor adapations

#### Motivation and Context
We need more sanity checks when pushing new commits in ferdi to avoid broken builds.

`eslint-plugin-unicorn` is a great, opinionated ruleset that catches errors and improves code flow.

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,8 @@
     "dev": "cross-env NODE_ENV=development gulp dev",
     "test": "jest",
     "test:watch": "jest --watch",
-    "lint": "eslint \"{src,scripts,packages,uidev}/**/*.{js,jsx,ts,tsx}\" --quiet --fix",
+    "lint": "eslint \"{src,scripts,packages,uidev}/**/*.{js,jsx,ts,tsx}\" --quiet",
```
Comment (vraravam) [inline package.json:28]: Do we need the `preserve-whitespace` flag here as well?

File: src/electron/ipc-api/download.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { PathLike } from 'fs';
 const debug = require('debug')('Ferdi:ipcApi:download');
 
 function decodeBase64Image(dataString: string) {
-  const matches = dataString.match(/^data:([A-Za-z-+/]+);base64,(.+)$/);
+  const matches = dataString.match(/^data:([+/A-Za-z-]+);base64,(.+)$/);
```
Comment (vraravam) [inline src/electron/ipc-api/download.ts:10]: Does this change cause a different behavior?

File: src/stores/GlobalErrorStore.js [hunk 3]
```diff
@@ -38,7 +38,7 @@ export default class GlobalErrorStore extends Store {
   }
 
   _handleConsoleError(type, error, url, line) {
-    if (typeof type === 'object' && type.length && type.length >= 1) {
+    if (typeof type === 'object' && type.length > 0 && type.length > 0) {
```
Comment (vraravam) [inline src/stores/GlobalErrorStore.js:41]: can we remove the duplication?

**Review by vraravam — COMMENTED** [review]
please see questions above

File: package.json (hunk 1 above)
Comment (vraravam) [reply package.json:28]: maybe this is what the conflict is about?

Comment (mhatvan) [reply package.json:28]: Might be, but I usually fix merge conflicts once I get the approvals in.

File: src/electron/ipc-api/download.ts (hunk 2 above)
Comment (mhatvan) [reply src/electron/ipc-api/download.ts:10]: Nope, but this was autofixed by: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/better-regex.md

They have a specific sort order for regex, that's why it moved to the front.

File: src/stores/GlobalErrorStore.js (hunk 3 above)
Comment (mhatvan) [reply src/stores/GlobalErrorStore.js]: Good catch, done in another commit.

**Review by vraravam — APPROVED** [review]

**Pull request merged**